src/common/transformations/src/transformations/fp16_compression/align_mixed_fp32_fp16_types.cpp

```diff
@@ -66,7 +66,7 @@ bool ov::pass::AlignMixedFP32FP16Types::run_on_model(const std::shared_ptr<ov::M
                         continue;
 
                     // todo xxx-101766: if we don't skip Results there is an error on GPU
-                    if (std::dynamic_pointer_cast<ov::op::v0::Result>(out_node))
+                    if (std::dynamic_pointer_cast<ov::op::v0::Result>(out_node) && !m_is_subgraph)
```

Comment:
maybe we can enable a test to below as well: src/common/transformations/tests/common_optimizations/align_mixed_fp32_fp16_types_test.cpp

Reply:
OK, will check this.